### DISCUSSION TITLE
Question about simulated projection data

### DISCUSSION THREAD

**Discussion by Olin1461** [astra-toolbox/astra-toolbox#384] [Q&A] [answered]
When using the Astra package to generate simulated projection data, do we assume that we are using a simple model for a monochromatic source of X-rays?

Thanks for your help!

**Comment by askorikov** [comment]
Hi! Yes indeed, a monochromatic model is assumed. But it's not difficult to extend it to simulating polychromatic data on the user side.

**Reply by Olin1461** [reply]
Thanks for your answer. Could you provide some hints on how to extend it to the polychromatic model? Should I perform the extension after obtaining the projection or before obtaining the projection matrix A?

Thanks!

**Reply by askorikov** [reply, marked as answer]
The approach would be to simulate monochromatic projections for a number of energy bins using material attenuations for corresponding energies, apply Beer-Lambert law and then weigh these projections by source-detector spectrum. Something like:
```python
I = np.zeros(projections_shape)
I0 = 0
for energy in energy_spectrum:
    attenuations = spectral_phantom(energy)
    source_intensity = source_spectrum(energy)
    detector_response = detector_spectrum(energy)
    I += source_intensity  * detector_response  * np.exp(-project_using_astra(attenuations))
    I0 += source_intensity  * detector_response
projections = -np.log(I/I0)
```

**Reply by Olin1461** [reply]
Thanks for your help!
